### PR TITLE
Revamp of build scripts

### PR DESCRIPTION
**Pull request by jskeet**
- Allow non-API packages to be specified, e.g.
  tagrelease.sh Google.Logging.V2 Google.Logging.Type
- Autodetect version number from project.json
- Copy nuget packages after build into releasebuild/nuget
- Remove pushrelease.sh as less relevant given the above change

### CODE REVIEW

File: tagrelease.sh [hunk 1]
```diff
@@ -22,43 +22,45 @@ then
...
 
 # Check that our master is up-to-date with upstream.
 # This will exit with a non-zero error code if the line isn't found.
 # TODO: Is this the best way to do this?
-git fetch -v --dry-run upstream 2>&1 \
-  | grep -E '\[up to date\]\s+master\s+->\s+upstream/master' > /dev/null \
-  || (echo Please update your local git repo before creating a release; exit 1)
+# git fetch -v --dry-run upstream 2>&1 \
+#  | grep -E '\[up to date\]\s+master\s+->\s+upstream/master' > /dev/null \
+#  || (echo Please update your local git repo before creating a release; exit 1)
```
Comment (chrisdunelm) [inline tagrelease.sh]: Are these meant to be here?

File: tagrelease.sh [hunk 2]
```diff
@@ -22,43 +22,45 @@ then
...
+# grep -E '"version": "[0-9]+\.[0-9]+\.[0-9]+(-[a-z0-9]+)?-\*"' apis/$api/$package/project.json 
+version=`grep -E '"version": "[0-9]+\.[0-9]+\.[0-9]+(-[a-z0-9]+)?-\*",' apis/$api/$package/project.json \
+   | sed -E 's/"version": "//g' \
+   | sed 's/-\*",//g' \
+   | sed 's/ //g'`
 
-commit=`git rev-parse master`
+echo "Creating release for package $package version $version"
+
+# commit=`git rev-parse master`
```
Comment (chrisdunelm) [inline tagrelease.sh]: Remove commented line?

File: tagrelease.sh [hunk 3]
```diff
@@ -22,43 +22,45 @@ then
...
 # TODO: Is this the best way to do this?
-git fetch -v --dry-run upstream 2>&1 \
-  | grep -E '\[up to date\]\s+master\s+->\s+upstream/master' > /dev/null \
-  || (echo Please update your local git repo before creating a release; exit 1)
+# git fetch -v --dry-run upstream 2>&1 \
+#  | grep -E '\[up to date\]\s+master\s+->\s+upstream/master' > /dev/null \
+#  || (echo Please update your local git repo before creating a release; exit 1)
 
-echo Creating release for API $api version $version
+# grep -E '"version": "[0-9]+\.[0-9]+\.[0-9]+(-[a-z0-9]+)?-\*"' apis/$api/$package/project.json
```
Comment (chrisdunelm) [inline tagrelease.sh]: Remove commented line?

**Comment by jskeet** [comment]
Whoops - thanks for catching the comments. Fixing (and squashing)...

**Pull request merged**